app/assets/javascripts/discourse/app/controllers/exception.js

```diff
@@ -51,6 +52,10 @@ export default Controller.extend({
   isServer: gte("thrown.status", 500),
   isUnknown: none("isNetwork", "isServer"),
 
+  // Handling for the detailed_404 setting (which actually creates 403s)
+  errorHtml: alias("thrown.responseJSON.extras.html"),
```

Comment:
I wonder if we need a CP for this. Might be simpler to omit and let the template use the `and` helper for `useErrorHtml` too.

Reply:
I couldn't quite figure out what you were saying here, so I just moved the `and` into the template, eliminating the `useErrorHtml` property. I kept the deep nesting alias.